src/Common/src/System/Globalization/IdnMapping.cs

```diff
@@ -98,71 +83,36 @@ public String GetAscii(String unicode, int index, int count)
             // We're only using part of the string
             unicode = unicode.Substring(index, count);
 
-            return GetAsciiUsingOS(unicode);
-        }
-
-        [System.Security.SecuritySafeCritical]
-        private String GetAsciiUsingOS(String unicode)
-        {
             if (unicode.Length == 0)
```

Comment:
So the actual difference is instead of throwing ArgumentException, we would throw a sub type (ArgumentOutOfRangeException) with a different message?

I think that's a reasonable change to make.  @davkean May have some thoughts from a compatibility standpoint.  The interesting question is that I suspect we would never port this change back to the Full .NET framework, so is the behavior difference between .NET Core and .NET 4.6 interesting in any way?

If it ends up not happening, it would not be the end of the world.  In that case perhaps we could keep the exception type but augment the message, if we thought it was really confusing.